tests/components/pooldose/test_config_flow.py

```diff
@@ -0,0 +1,433 @@
+"""Test the Seko Pooldose config flow."""
+
+from unittest.mock import AsyncMock, MagicMock, patch
+
+from pooldose.request_status import RequestStatus
+import pytest
+
+from homeassistant.components.pooldose.const import DOMAIN
+from homeassistant.config_entries import SOURCE_USER
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from tests.common import MockConfigEntry
+
+
+@pytest.fixture
+def mock_pooldose_client():
+    """Create a mock PooldoseClient."""
+    client = MagicMock()
+    client.device_info = {"SERIAL_NUMBER": "SN123456789"}
+    client.connect = AsyncMock(return_value=RequestStatus.SUCCESS)
+    client.check_apiversion_supported = MagicMock(
+        return_value=(RequestStatus.SUCCESS, {})
+    )
+    client.is_connected = True
+    return client
```

Comment:
Instead, create a `conftest.py` and create mock that mocks both your config flow object as the normal object. For example, check `overseerr`

```py
@pytest.fixture
def mock_overseerr_client() -> Generator[AsyncMock]:
    """Mock an Overseerr client."""
    with (
        patch(
            "homeassistant.components.overseerr.coordinator.OverseerrClient",
            autospec=True,
        ) as mock_client,
        patch(
            "homeassistant.components.overseerr.config_flow.OverseerrClient",
            new=mock_client,
        ),
    ):
        client = mock_client.return_value
        client.get_request_count.return_value = RequestCount.from_json(
            load_fixture("request_count.json", DOMAIN)
        )
        yield client
```